### ISSUE TITLE
Generate pull request title from issue title

### ISSUE THREAD

**Issue by Piedone** [integrations/jira#351]
The current "Create pull request" button available under a JIRA issue will create a GitHub pull request where the PR's title will be generated from the branch name. E.g. for a branch named "issue/KEY-123" the PR title will be "Issue/key 123" (not the different casing and lack of dash).

Instead of this (possible as an option), the PR's title could come from the title of the issue, concatenated with its issue key. E.g. "KEY-123 Implementing awesome feature". This way we'd a) get an actual legible title for the PR right off the bat instead of having to edit it all the time and b) the PR would be connected to the JIRA issue (I think unless the issue key is mentioned in the PR title it won't show up under the issue, just the branch).

As an added super-awesomeness the body of the PR could be auto-populated too, also referencing the JIRA issue. This could be something like "Click here for the corresponding JIRA issue." that would link to the issue. This way you could not just reach the PR from the issue easily but also the other way around.

**Comment by Berkmann18** [comment]
That would be nice although, having a similar system (based on GitLab) at work, it makes it easier to see what ticket is linked to PR/MR, but at the same time, if you don't know what the ticket entails or read the PR/MR summary then it wouldn't be clear off the bat.

**Comment by Piedone** [comment]
The issue's title should give you an idea of what the PR is about too. In the cases it doesn't you could of course still manually edit it.

**Comment by Berkmann18** [comment]
True, I was kind of playing devil's advocate here having worked in teams where PR had meaningful titles based on their nature/change and in GitLab(+Jira) where MRs referred to tickets (e.g. `Feat/proj 124`) both have their use.

**Comment by rachellerathbone** [comment]
Hi @Piedone. I'm from a team at Atlassian that is currently in the process of migrating the ownership over to us from GitHub. As a part of this, we're currently going through the backlog of issues and prs to see what can still be addressed and what should be closed.

I've flagged this with the 'to triage' label so my team can discuss. We'll keep you posted on the status of this.

**Comment by Piedone** [comment]
I see, thanks!

**Comment by rachellerathbone** [comment]
Hi @Piedone. Good news, we are now actively looking into solutions for this. Stay tuned!

**Comment by Piedone** [comment]
Oh that's fantastic, thanks!

**Comment by tomasfejfar** [comment]
This would be great! Currently we're doing all this manually.

**Comment by tzachshabtay** [comment]
Hi @rachellerathbone, 2.5 years later and wondering if there are any updates? We just migrated to Jira from Shortcut which has that feature built in.

**Comment by oceanofmaya** [comment]
Any updates on this issue? This adds unnecessary manual correction each time if autolinking to JIRA is desired from merge commits.
